accelerator/http.c

```diff
@@ -8,18 +10,110 @@ typedef struct ta_http_request_s {
   char *request;
 } ta_http_request_t;
 
+static status_t ta_http_url_matcher(char const *const url,
+                                    char *const regex_rule) {
+  if (url == NULL || regex_rule == NULL) {
+    return SC_HTTP_NULL;
+  }
+  regex_t reg;
+  status_t ret = SC_OK;
+  int reg_flag = REG_EXTENDED | REG_NOSUB;
+
+  if (regcomp(&reg, regex_rule, reg_flag) != 0) {
+    return SC_HTTP_INVALID_REGEX;
+  }
+  if (regexec(&reg, url, 0, NULL, 0) != 0) {
+    // Did not match pattern
+    ret = SC_HTTP_URL_NOT_MATCH;
+  }
+
+  regfree(&reg);
+  return ret;
+}
+
+static status_t ta_get_url_parameter(char const *const url, int index,
+                                     char **param) {
+  if (url == NULL || param == NULL) {
+    return SC_HTTP_NULL;
+  }
+  // Copy url for parsing
+  int url_len = strlen(url);
```

Comment:
should we check out `url_len > 0` here?

Reply:
Great point! I didn't think of this.
Yet the url would be `/` if the request url is `http://localhost:8000`, thus we don't need to worry about the `url_len`. Furthermore, I think this check `url == NULL` can be removed.

Reply:
I'll also add checks on `index`